janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/inmemory/InMemoryKeyColumnValueStore.java

```diff
@@ -104,6 +124,429 @@ public void close() throws BackendException {
+
+                        for (i = 0; i < reductionLevels.length; i++) {
+                            if (fr.getAchievablePageReduction() <= reductionLevels[i]) {
+                                break;
+                            }
+                        }
+                        reductionCounts[i]++;
+                    }
+                }
+
+            }
+        }
+
+        FragmentationReport report = new FragmentationReport();
+        report.setName(name);
+        report.setNumStores(kcv.size());
+        report.setNumMultipageStores(numMultipageStores);
+        report.setTotalPageCount(totalPageCount);
+        report.setNumFragmentedStores(numFragmentedStores);
+        report.setTotalFragmentedPages(totalFragmentedPages);
+        report.setNumCompressableStores(numCompressableStores);
+        report.setTotalCompressablePages(totalCompressablePages);
+        report.setTotalAchievablePageReduction(totalAchievablePageReduction);
+        report.setKeysByteSize(keysByteSize);
+
+        report.setEntryLevels(entryLevels);
+        report.setEntryCounts(entryCounts);
+
+        report.setPageLevels(pageLevels);
+        report.setPageCounts(pageCounts);
+
+        report.setCompressablePageLevels(compressablePageLevels);
+        report.setCompressablePageCounts(compressablePageCounts);
+
+        report.setReductionLevels(reductionLevels);
+        report.setReductionCounts(reductionCounts);
+
+        report.setStoresToDefragment(storesToDefragment);
+
+        return report;
+    }
+
+    public void quickDefragment(Collection<ColumnValueStore> stores, StoreTransaction txh) throws BackendException {
+        for (ColumnValueStore cvs : stores) {
+            cvs.quickDefragment(txh);
+        }
+    }
+
+    public void quickDefragment(StoreTransaction txh) throws BackendException {
+        quickDefragment(kcv.values(), txh);
+    }
+
+    private static OutputStream compressedOutputStream(OutputStream streamToWrap) {
+        return new DeflaterOutputStream(streamToWrap, new Deflater(BEST_SPEED, true), true);
+    }
+
+    private static InputStream compressedInputStream(InputStream streamToWrap) {
+        return new InflaterInputStream(streamToWrap, new Inflater(true));
+    }
+
+    public void dumpTo(Path storePath, ForkJoinPool parallelOperationsExecutor) {
+        if (kcv.size() < 1)
+            return;
+
+        int numChunks = Runtime.getRuntime().availableProcessors() * 2;
+
+        int chunkSize = kcv.size() > 1000 ? kcv.size() / numChunks : kcv.size();
+
+        ArrayList<List<Map.Entry<StaticBuffer, ColumnValueStore>>> chunks = Lists.newArrayList(Iterators.partition(kcv.entrySet().iterator(), chunkSize));
+
+        IntStream.range(0, chunks.size()).mapToObj(i ->
+        {
+            Path filePath = Paths.get(storePath.toString(), getName() + "_" + i);
+
+            return parallelOperationsExecutor.submit(() -> dumpChunk(filePath, chunks.get(i)));
+        }).collect(Collectors.toList())
```

Comment:
why is this collect operation necessary? you stream right after it,

Reply:
it is necessary to ensure that the preceding map() is executed eagerly and all tasks are actually submitted to execute in parallel.
without the collect, it would execute lazily, submit tasks one by one and immediately block until it is done - so beating the purpose of executing them on the thread pool in the first place... 
will add a comment to this effect